nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -862,31 +866,30 @@ contract StakingEscrow is Issuer {
         uint16 nextPeriod = currentPeriod.add16(_periods);
         result = new address[](_points.length);
 
+        uint256 previousPoint = 0;
         uint256 pointIndex = 0;
         uint256 sumOfLockedTokens = 0;
         uint256 stakerIndex = 0;
-        bool addMoreTokens = true;
         while (stakerIndex < stakers.length && pointIndex < _points.length) {
             address currentStaker = stakers[stakerIndex];
             StakerInfo storage info = stakerInfo[currentStaker];
-            uint256 point = _points[pointIndex];
             if (info.confirmedPeriod1 != currentPeriod &&
                 info.confirmedPeriod2 != currentPeriod) {
                 stakerIndex += 1;
-                addMoreTokens = true;
                 continue;
             }
-            if (addMoreTokens) {
-                sumOfLockedTokens = sumOfLockedTokens.add(getLockedTokens(info, currentPeriod, nextPeriod));
-            }
-            if (sumOfLockedTokens > point) {
+            uint256 stakerTokens = getLockedTokens(info, currentPeriod, nextPeriod);
+            uint256 nextSumValue = sumOfLockedTokens.add(stakerTokens);
```

Comment:
Interesting. So, instead of post-selecting points which don't belong to one staker, you switch to the next staker if you selected a point. I think, it makes sense (in the limit of `stake << point_distance`), so I'm ok to do it this way. But also good run the simulations you had

Reply:
I'm not 100% convinced though that it won't distort the probability distribution too much. It won't break things, so I am +1 on merging. But need to do tests for sure (comparing it with the situation when sample can select multiple points per staker